src/drafter.h

```diff
@@ -1,37 +1,101 @@
+//  Copyright (c) 2016 Apiary Inc. All rights reserved.
 //
 
-#ifndef DRAFTER_H
-#define DRAFTER_H
+#ifndef DRAFTER_API_H
+#define DRAFTER_API_H
 
-#include "snowcrash.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
 
+//#if defined(BUILDING_DRAFTER)
+#if defined(BUILDING_SNOWCRASH)
+#   define DEPRECATED
+#endif
 
-/**
- *  API Blueprint Parser Interface
- *  ------------------------------
- *
- *  This is the parser's entry point.
- */
+#if defined(_MSC_VER)
+#   if !defined(DEPRECATED)
+#       define DEPRECATED __declspec(deprecated)
+#   endif
+#elif defined(__clang__) || defined(__GNUC__)
+#   if !defined(DEPRECATED)
+#       define DEPRECATED __attribute__((deprecated))
+#   endif
+#else
+#   if !defined(DEPRECATED)
+#       define DEPRECATED
+#   endif
+#endif
 
-namespace drafter {
+#ifndef DRAFTER_API
+#  if defined _WIN32 || defined __CYGWIN__
+#     if defined(DRAFTER_BUILD_SHARED) /* build dll */
+#         define DRAFTER_API __declspec(dllexport)
+#     elif !defined(DRAFTER_BUILD_STATIC) /* use dll */
+#         define DRAFTER_API __declspec(dllimport)
+#     else /* static library */
+#         define DRAFTER_API /* nothing */
+#     endif
+#  else
+#     if __GNUC__ >= 4
+#         define DRAFTER_API __attribute__((visibility("default")))
+#     else
+#         define DRAFTER_API
+#     endif
+#  endif
+#endif
 
-    /**
-     *  \brief Parse the source data into a blueprint abstract source tree (AST).
-     *
-     *  \param source       A textual source data to be parsed.
-     *  \param options      Parser options. Use 0 for no additional options.
-     *  \param out          Output buffer to store parsing result into.
-     *  \return Error status code. Zero represents success, non-zero a failure.
-     */
-    int ParseBlueprint(const mdp::ByteBuffer& source,
-                       snowcrash::BlueprintParserOptions options,
-                       const snowcrash::ParseResultRef<snowcrash::Blueprint>& out);
+
+#ifndef __cplusplus
+typedef struct drafter_result drafter_result;
+#else
+namespace refract {
+    struct IElement;
+}
+typedef refract::IElement drafter_result ;
+#endif
+
+
+typedef enum {
+    DRAFTER_SERIALIZE_YAML = 0,
+    DRAFTER_SERIALIZE_JSON
+} drafter_format;
+
+typedef struct {
+    bool sourcemap;
+    drafter_format format;
+} drafter_options;
+
+/* Parse API Bleuprint and serialize it to given format.*/
+DRAFTER_API int drafter_parse_blueprint_to(const char* source,
+                               const char ** out,
+                               drafter_options options);
+
+/* Parse API Bleuprint and return result, which is a opaque handle for
+ * later use*/
+DRAFTER_API int drafter_parse_blueprint(const char* source, drafter_result** out);
+
+/* Serialize result to given format*/
+DRAFTER_API const char* drafter_serialize(drafter_result *res, drafter_options options);
+
+/* Free memory alocated for result handler */
+DRAFTER_API void free_drafter_result(drafter_result* result);
```

Comment:
I feel like this function should be called `drafter_free_result` so it has the same prefix as the others. Drafter keyword is always at the start. This also allows users to "tab"-complete drafter functions consistently.

Any thoughts?